File: src/eShop.Data/Entities/Categories/Category.cs
using eShop.Data.Entities.Products;

namespace eShop.Data.Entities.Categories
{
    public class Category
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public string? Description { get; set; }
        public ICollection<Product> Products { get; set; } = new List<Product>();
        public ICollection<CategoryAttribute> Attributes { get; set; } = new List<CategoryAttribute>();
    }
}

File: src/eShop.Admin/src/shared/types/category.ts
import { Product } from "./product";

export type Category = {
    id: number;
    name: string;
    description: string;
}

export type CategoryDetail = {
    id: number;
    name: string,
    description: string;
    attributes: CategoryAttribute[],
    products: Product[],
}

export type CategoryAttribute = {
    id: number;
    name: string;
    displayName: string;
}